apps/challenges/views.py

```diff
@@ -777,6 +839,87 @@ def create_challenge_using_zip_file(request, challenge_host_team_pk):
         response_data = {"error": message}
         return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
 
+    # Manipulating the challenge data of the template challenge with data sent by hosts
+    if is_template_challenge:
+        yaml_file_data["title"] = challenge_data_from_hosts.get("title")
+        yaml_file_data["short_description"] = challenge_data_from_hosts.get(
+            "short_description"
+        )
+        yaml_file_data["description"] = challenge_data_from_hosts.get(
+            "description"
+        )
+        yaml_file_data["evaluation_details"] = challenge_data_from_hosts.get(
+            "evaluation_details"
+        )
+        yaml_file_data["image"] = challenge_data_from_hosts.get("image")
+        yaml_file_data[
+            "submission_guidelines"
+        ] = challenge_data_from_hosts.get("submission_guidelines")
+        yaml_file_data[
+            "leaderboard_description"
+        ] = challenge_data_from_hosts.get("leaderboard_description")
+        yaml_file_data["evaluation_script"] = challenge_data_from_hosts.get(
+            "evaluation_script"
+        )
+        yaml_file_data["start_date"] = challenge_data_from_hosts.get(
+            "start_date"
+        )
+        yaml_file_data["end_date"] = challenge_data_from_hosts.get("end_date")
+        yaml_file_data["leaderboard"] = json.load(
+            challenge_data_from_hosts.get("leaderboard")
+        )
+
+        # Mapping the challenge phase data to that in yaml_file_data
+        challenge_phases_data = yaml_file_data["challenge_phases"]
+        challenge_phases_data_from_hosts = challenge_data_from_hosts.get(
+            "challenge_phases"
+        )
+        for i in range(len(challenge_phases_data)):
+            challenge_phases_data[i][
+                "name"
+            ] = challenge_phases_data_from_hosts.get("challenge_phases")[
+                i
+            ].get(
+                "name"
```

Comment:
same here. Use an array of field names to loop through and override values if present in request